browser/core_metrics/core_metrics_service_factory.cc

```diff
@@ -0,0 +1,55 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/core_metrics/core_metrics_service_factory.h"
+#include "brave/components/core_metrics/core_metrics_service.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/history/history_service_factory.h"
+#include "chrome/browser/profiles/incognito_helpers.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/history/core/browser/history_service.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+
+namespace core_metrics {
+
+// static
+CoreMetricsServiceFactory* CoreMetricsServiceFactory::GetInstance() {
+  return base::Singleton<CoreMetricsServiceFactory>::get();
+}
+
+// static
+CoreMetricsService* CoreMetricsServiceFactory::GetServiceForContext(
+    content::BrowserContext* context) {
+  if (context->IsOffTheRecord()) {
```

Comment:
Should be moved to `GetBrowserContextToUse`
Please use this pattern: https://source.chromium.org/chromium/chromium/src/+/main:components/keyed_service/content/browser_context_keyed_service_factory.cc;l=61-68;drc=21a4c84dd469bfdc9f27375133dc309b62101ef8